core/src/main/java/org/elasticsearch/index/mapper/ip/IpFieldMapper.java

```diff
@@ -206,6 +209,23 @@ public BytesRef indexedValueForSearch(Object value) {
         }
 
         @Override
+        public Query termQuery(Object value, @Nullable QueryShardContext context) {
+            if (value != null) {
+                long[] fromTo;
+                if (value instanceof BytesRef) {
+                    fromTo = InetAddresses.cidrMaskToMinMax(((BytesRef) value).utf8ToString());
+                } else {
+                    fromTo = InetAddresses.cidrMaskToMinMax(value.toString());
+                }
+                if (fromTo != null) {
+                    return rangeQuery(fromTo[0] < 0 ? Double.NEGATIVE_INFINITY : fromTo[0],
+                            fromTo[1] < 0 ? Double.POSITIVE_INFINITY : fromTo[1], true, false);
```

Comment:
can you use `null` instead of POSITIVE/NEGATIVE_INFINITY? this is the way the rangeQuery method expects to be notified about open bounds